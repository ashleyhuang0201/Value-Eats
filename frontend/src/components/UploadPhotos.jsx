import React from 'react';
import { Box } from '@material-ui/core';
import AddAPhotoIcon from '@material-ui/icons/AddAPhoto';
import { fileToDataUrl } from '../utils/helpers';
import { ImagePreview } from '../styles/ImagePreview';
import { Label } from '../styles/Label';
import { FileUpload } from '../styles/FileUpload';

export default function UploadPhotos ({ setImages, previewImages, setPreviewImages, uploadDescription }) {

  const handleImages = (data) => {
    setImages([]);
    Array.from(data).forEach((file) => {
        fileToDataUrl(file).then((url) => {
            setImages((prevArray) => [...prevArray, url]);
        });
    });

    if (data) {
        const fileArray = Array.from(data).map((file) =>
            URL.createObjectURL(file)
        );
        setPreviewImages(fileArray);
    }
  };

  const getPreviewImages = (data) => {
    return data.map((photo) => {
        return <ImagePreview src={photo} key={photo} />;
    });
  };

  return (
    <>
      <Box pt={2}>
        <Label>
            <FileUpload
                type="file"
                multiple
                onChange={(e) => handleImages(e.target.files)}
            />
            {<AddAPhotoIcon />} {uploadDescription}
        </Label>
      </Box>
      <Box flex-wrap="wrap" flexDirection="row" width="60%">
          {getPreviewImages(previewImages)}
      </Box>
    </>
  )
}