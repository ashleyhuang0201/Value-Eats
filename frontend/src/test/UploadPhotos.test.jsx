import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import UploadPhotos from "../components/UploadPhotos";

describe("UploadPhotos", () => {
  const noop = () => {};

  it("can work in the base case", () => {
    const upload = shallow(<UploadPhotos setImages={noop} previewImages={[]} setPreviewImages={noop} uploadDescription={"Upload"}/>);
    // console.log(upload.debug({ verbose: true }));
    expect(upload.exists()).toBeTruthy();
  });

  it("text is on the upload element", () => {
    const upload = shallow(<UploadPhotos setImages={noop} previewImages={[]} setPreviewImages={noop} uploadDescription={"Upload"}/>);
    console.log(upload.debug({ verbose: true }));
    expect(upload.text()).toBe("Upload");
  });

  // Snapshot tests
  it("renders with minimal props", () => {
    const rating = renderer.create(<UploadPhotos setImages={noop} previewImages={[]} setPreviewImages={noop} uploadDescription={"Upload"}/>);
    expect(rating).toMatchSnapshot();
  });

  it("renders with images", () => {
    const previewImages = [{ img: "", id: 0 }, { img: "", id: 1 }];
    const upload = renderer.create(<UploadPhotos setImages={noop} previewImages={previewImages} setPreviewImages={noop} uploadDescription={"Upload"}/>);
    expect(upload).toMatchSnapshot();
  });
});
