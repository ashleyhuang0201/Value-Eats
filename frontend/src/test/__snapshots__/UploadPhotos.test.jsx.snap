// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPhotos renders with images 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <label
      className="sc-gtsrHT kWETdg"
    >
      <input
        accept="image/png, image/jpg, image/jpeg"
        className="sc-dlnjwi gVzoKn"
        multiple={true}
        onChange={[Function]}
        type="file"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
        />
      </svg>
      Upload
    </label>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-4"
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "bottom": "35px",
            "left": "35px",
            "position": "absolute",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </span>
      </button>
      <img
        className="sc-bdnxRM dUZkKx"
        src={
          Object {
            "id": 0,
            "img": "",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "bottom": "35px",
            "left": "35px",
            "position": "absolute",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
            />
          </svg>
        </span>
      </button>
      <img
        className="sc-bdnxRM dUZkKx"
        src={
          Object {
            "id": 1,
            "img": "",
          }
        }
      />
    </div>
  </div>,
]
`;

exports[`UploadPhotos renders with minimal props 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <label
      className="sc-gtsrHT kWETdg"
    >
      <input
        accept="image/png, image/jpg, image/jpeg"
        className="sc-dlnjwi gVzoKn"
        multiple={true}
        onChange={[Function]}
        type="file"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
        />
      </svg>
      Upload
    </label>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-2"
  />,
]
`;
